povm_toolbox/quantum_info/base_dual.py

```diff
@@ -0,0 +1,53 @@
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""TODO."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+import numpy as np
+from qiskit.quantum_info import Operator
+
+from .base_frame import BaseFrame
+from .base_povm import BasePOVM
+
+
+class BaseDUAL(BaseFrame, ABC):
+    """Abstract base class that contains all methods that any specific DUAL subclass should implement."""
+
+    @property
+    def n_outcomes(self) -> int:
+        """Give the number of outcomes of the POVM."""
```

Comment:
```suggestion
        """Give the number of outcomes of the DUAL."""
```
Not sure.. Did you want to explicitly link to the DUAL's POVM here?